pulp_deb/app/viewsets/content.py

```diff
@@ -36,11 +40,136 @@ class GenericContentViewSet(SingleArtifactContentUploadViewSet):
+    conjunction with a new filter that naively follows the foreign key references in the database:
+    packages.filter(deb_packagereleasecomponent__release_component__release=release_uuid)
+
+    What Django does if you call two separate filters is use the first to filter the QuerySet,
+    then use the second to filter the QuerySet further. This is *different* than calling
+    filter once with both conditions.
+    https://docs.djangoproject.com/en/dev/topics/db/queries/#spanning-multi-valued-relationships
+
+    Example: packages.filter("in RepositoryVersion2").filter("in Release1")
+    This will return both Package1 and Package2, which is not what we wanted. In the first filter it
+    looks and says "yep, both Package1 and Package2 are in RepositoryVersion2", and then the second
+    filter is applied and it says "yep, both Package1 and Package2 were in Release1 at some point".
+    This is because the linkage via PackageReleaseComponent2 still *exists*, it's just not in
+    RepositoryVersion2.
+
+    What we really _actually_ want is to apply _both_ conditions to the PackageReleaseComponent
+    mapping as an intermediate step, so both release_uuid and repository_version_href must be
+    passed to our new filter:
+    packages.filter(package.PRC in PRC.filter("in RepositoryVersion2", "in Release1"))
+
+    This guarantees that we are only considering Packages with both requirements, and returns only
+    Package1.
+    """
+
+    HELP = "Override with your value-specific help message"
+    ARG = "Override with the type of your arg, for example package_href"
+    GENERIC_HELP = """
+    Must be a comma-separated string: "{arg},repository_or_repository_version_href"
+    {arg}: {help}
+    repository_or_repository_version_href: The RepositoryVersion href to filter by, or Repository
+        href (assume latest version)
+    """
+
+    def __init__(self, *args, **kwargs):
+        kwargs.setdefault(
+            "help_text", _(self.GENERIC_HELP).format(arg=_(self.ARG), help=_(self.HELP))
+        )
+        super().__init__(*args, **kwargs)
+
+    def filter(self, qs, value):
+        """
+        Args:
+            qs (django.db.models.query.QuerySet): The Content Queryset
+            value (string, "value,repository_version_href"): The values to filter by
+        """
+        if value is None:
+            # user didn't supply a value
+            return qs
+
+        repo_version = None
+        my_value, r_or_rv_href = value.split(",", 1)
+        if not my_value or not r_or_rv_href or "," in r_or_rv_href:
+            raise ValidationError(detail=_("Invalid value supplied for content filter"))
+
+        repo_version = NamedModelViewSet.get_resource(r_or_rv_href)
+        if isinstance(repo_version, Repository):
+            repo_version = repo_version.latest_version()
+
+        if not isinstance(repo_version, RepositoryVersion):
+            raise ValidationError(detail=_("Invalid value supplied for content filter"))
```

Comment:
Should we try to give more detailed hints here on what went wrong? e.g:
```suggestion
            raise ValidationError(detail=_("Could not resolve a repository version from value supplied for content filter"))
```
At the least this would allow anyone debugging to easily see which of the two ValidationError the user ran into.

Reply:
Done.